src/main/java/com/teragrep/aer_01/DefaultOutput.java

```diff
@@ -0,0 +1,189 @@
+final class DefaultOutput implements Output {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultOutput.class);
+
+    private final RelpConnection relpConnection;
+    private final String relpAddress;
+    private final int relpPort;
+    private final int reconnectInterval;
+
+    // metrics
+    private final Counter records;
+    private final Counter bytes;
+    private final Counter resends;
+    private final Counter connects;
+    private final Counter retriedConnects;
+    private final Timer sendLatency;
+    private final Timer connectLatency;
+
+
+    DefaultOutput(
+            String name,
+            RelpConfig relpConfig,
+            MetricRegistry metricRegistry) {
+        this.relpAddress = relpConfig.destinationAddress;
+        this.relpPort = relpConfig.destinationPort;
+        this.reconnectInterval = relpConfig.reconnectInterval;
+
+        this.relpConnection = new RelpConnection();
+        this.relpConnection.setConnectionTimeout(relpConfig.connectionTimeout);
+        this.relpConnection.setReadTimeout(relpConfig.readTimeout);
+        this.relpConnection.setWriteTimeout(relpConfig.writeTimeout);
+
+        this.records = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "records"));
+        this.bytes = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "bytes"));
+        this.resends = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "resends"));
+        this.connects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "connects"));
+        this.retriedConnects = metricRegistry.counter(name(DefaultOutput.class, "<[" + name + "]>", "retriedConnects"));
+        this.sendLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "sendLatency"));
+        this.connectLatency = metricRegistry.timer(name(DefaultOutput.class, "<[" + name + "]>", "connectLatency"));
+
+        connect();
+    }
+
+    private void connect() {
+        boolean connected = false;
+        while (!connected) {
+            try (final Timer.Context context = connectLatency.time()) {
+                connected = this.relpConnection.connect(relpAddress, relpPort);
+                connects.inc();
+            } catch (IOException | TimeoutException e) {
+                LOGGER.error("Exception while connecting to <[{}]>:<[{}]>", relpAddress, relpPort, e);
+            } catch (UnresolvedAddressException e) {
+                LOGGER.error("Can't resolve address of target <[{}]>", relpAddress, e);
+            }
+
+            if (!connected) {
+                try {
+                    Thread.sleep(reconnectInterval);
+                    retriedConnects.inc();
+                } catch (InterruptedException e) {
+                    LOGGER.warn("Sleep interrupted while waiting for reconnectInterval <{}> on <[{}]>:<[{}]>", reconnectInterval, relpAddress, relpPort, e);
```

Comment:
reconnectInterval is user supplied variable so it must be logged with brackets: <[{}]>